### DISCUSSION TITLE
Equivalent for openmct preview actions?

### DISCUSSION THREAD

**Discussion by akhna10x** [nasa/openmct#4067] [Community Q&A] [answered]
Hi, 

Is there a way to create/open a plugin as a preview window in OpenMCT 1.7?

We're trying to migrate from a pre 1.0 version, where we'd use mct-preview-action to open detail views of given components. 

For example, to open a preview window from a larger list of telemetry objects, we'd use:

`
   openmct.$injector.get('objectService').getObjects([key])
      .then((objs) => {
        objs[key].getCapability('action')
          .getActions({ key: 'mct-preview-action' })[0]
          .perform()
      })
`

Is there a native way to open a preview window in the latest version of OpenMCT (this would be opening a separate plugin, rather than an enlarged view of a widget on the page)?

Any insight into this would be greatly appreciated.

**Comment by akhenry** [comment, marked as answer]
If I'm understanding the question correctly you want to open a view of an object that you've created in a plugin in a Preview window. You were able to do this using the legacy API using the code that you provided.

The equivalent of the code that you've provided in our current API would be this:

``` JavaScript
const objectPath = await openmct.objects.getOriginalObjectPath({namespace: `${namespace}`, key: `${key}`});
const previewAction = openmct.actions.getAction('preview');

previewAction.invoke(objectPath);
```

The `namespace` and `key` come from the object id. In legacy Open MCT object ids were strings of the form `"namespace:key"`, but in our current API objects are uniquely identified by an "identifier" object of the form `{namespace: <string>, key: <string>}`. 

eg. the legacy object id `"spacecraft:voltage"` would be `{namespace: "spacecraft", key: "voltage"}` in our current API.

**Reply by akhna10x** [reply]
Great, thank you, I haven't tried this out quite yet but that sounds like exactly what we need.
